cdk/domino_cdk/config/template.py

```diff
@@ -28,6 +29,9 @@ def config_template(
         disk_size = 100
         platform_instance_type = "m5.4xlarge"
 
+    if istio_compatible:
```

Comment:
Dunno if this is less clear / emphasizes the wrong things, but could save a line and make this `if dev_defaults or istio_compatible:`

Reply:
Going to leave it as-is for clarity for now

Reply:
I really do not like this
```
do
if stuff, undo and do differently
```
style

For me it is 3 different configs and 2 bits to chose one.

So I would write
```
normal_config = {
    max_nodegroup_azs = 3
    destroy_on_destroy = False
    disk_size = 1000
    platform_instance_type = "m5.2xlarge"
}

dev_config = {
        max_nodegroup_azs = 1
        destroy_on_destroy = True
        disk_size = 100
        platform_instance_type = "m5.4xlarge"
  }

normal_config_with_istio = {
    max_nodegroup_azs = 3
    destroy_on_destroy = False
    disk_size = 1000
    platform_instance_type = "m5.4xlarge"
}

if not dev and not istio:
  config = normal_config
elif not dev and istio:
  config = normal_config_with_istio
else:
  config = dev_config
```

code to mutate something already done in the previous lines to save a line is not so readable. If you do this in scala, you would probably use val to make stuff immutable and match to select one of the values.

Reply:
It is like "common law". You have original law and you have a myriad of amendments. In the end you need a lawyer to get through.

Reply:
Note that here it is not a "builder" pattern, where you start with something and then add stuff.

Reply:
Updated :+1: